### PR TITLE
userland: remove Makefile-app

### PR DESCRIPTION
**Pull request by bradjc**
### Pull Request Overview

Makefile-app won't exist with userland in its own repo, so this moves the programming logic to a makefile within userland.

### Testing Strategy

This pull request was tested by running `make TOCK_BOARD=x program`.

### TODO or Help Wanted

Is there a better way to do this?

### Documentation Updated

- [ ] Kernel: Updated the relevant files in `/docs`, or no updates are required.
- [ ] Userland: Added/updated the application README, if needed.

### Formatting

- [ ] Ran `make formatall`.

### CODE REVIEW

File: userland/Program.mk [hunk 1]
```diff
@@ -3,18 +3,46 @@
...
+# Change program region offset
+TOCKLOADER_INSTALL_FLAGS += --app-address 0x40000
+PROGRAM = $(TOCKLOADER) $(TOCKLOADER_GENERAL_FLAGS) install $(TOCKLOADER_INSTALL_FLAGS) $<
+FLASH = $(TOCKLOADER) $(TOCKLOADER_GENERAL_FLAGS) install $(TOCKLOADER_INSTALL_FLAGS) --jtag $<
+
+else ifeq ("$(TOCK_BOARD)","ek-tm4c1294xl")
+FLASH = $(OPENOCD) -c "source [find board/ek-tm4c1294xl.cfg]; init; reset halt; flash write_image erase $< 0x00020000 bin; reset; shutdown"
+
+else ifeq ("$(TOCK_BOARD)","nrf51dk")
+FLASH = $(TOCK_USERLAND_BASE_DIR)/tools/program/nrf51dk.py $<
```
Comment (niklasad1) [inline userland/Program.mk]: Should we remove `tools/program/nrf5x.py` and use tockloader instead?

Comment (brghena) [reply userland/Program.mk]: We absolutely should support the nRF51 and 52 in tockloader. Does that support exist right now?

Comment (bradjc) [reply userland/Program.mk]: I use tockloader for the nrf52. `tockloader install --jlink --board nrf52dk`

Comment (niklasad1) [reply userland/Program.mk]: Yepp, we support both but I think we should aim to use tockloader as much as possible and reduce the amount of board-specific scripts if that is possible!

However ```make TOCK_BOARD=nrf52dk``` is using userland/tools/program/nrf52dk.py under the hood!

Comment (brghena) [reply userland/Program.mk]: Having chatted with Brad about this a little out-of-band, I think we should move the nRF platforms over to Tockloader by default as a separate pull request. It's a good thing to do, but not really part of this change.

**Pull request merged**